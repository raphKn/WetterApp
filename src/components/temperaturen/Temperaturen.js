import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import Temperatur from './Temperatur';
import CustomSection from '../common/CustomSection';
import { isEmptyObject } from '../../utils/ObjectUtils';

const styles = StyleSheet.create({
    image: { width: 70, height: 70 },
    temperaturenContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    temperaturenHeader: {
        fontSize: 24,
    },
});

const Temperaturen = (props) => {
    if (isEmptyObject(props.temperaturen)) {
        return null;
    }
    return (<View>
        <CustomSection style={styles.temperaturenContainer} name="Temperaturen">
            <Temperatur value={props.temperaturen.temperatur} />
            <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
                <Temperatur value={props.temperaturen.mindestTemperatur} isMinTemp />
                <Temperatur value={props.temperaturen.maximaleTemperatur} isMaxTemp />
            </View>
        </CustomSection>
    </View>);
};

Temperaturen.propTypes = ({
    temperaturen: PropTypes.shape({
        temperatur: PropTypes.string,
        mindestTemperatur: PropTypes.string,
        maximaleTemperatur: PropTypes.string,
    }),
});

Temperaturen.defaultProps = ({
    temperaturen: null,
});

export default Temperaturen;
