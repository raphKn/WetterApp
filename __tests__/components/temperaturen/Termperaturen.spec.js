import React from 'react';
import renderer from 'react-test-renderer';
import Temperaturen from '../../../src/components/temperaturen/Temperaturen';

describe('Temperaturen', () => {
    it('renders correctly in a snapshot', () => {
        const temperaturen = { temperatur: '12', mindestTemperatur: '10', maximaleTemperatur: '14' };
        const wrapper = renderer.create(
            <Temperaturen temperaturen={temperaturen} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders null for no temperaturen', () => {
        const temperaturen = {};
        const wrapper = renderer.create(
            <Temperaturen temperaturen={temperaturen} />);
        expect(wrapper).toMatchSnapshot();
    });
});
