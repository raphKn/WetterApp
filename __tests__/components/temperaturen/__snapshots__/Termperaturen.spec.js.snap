// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperaturen renders correctly in a snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#3770a5",
        },
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#FAFAFA",
            },
            Object {
              "fontSize": 18,
              "paddingHorizontal": 18,
            },
          ]
        }
      >
        Temperaturen
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E0E0E0",
            "paddingHorizontal": 18,
            "paddingVertical": 6,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "color": undefined,
                "fontSize": 32,
              },
            ]
          }
        >
          12
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          aktuell
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#212121",
                },
                Object {
                  "color": "blue",
                  "fontSize": 32,
                },
              ]
            }
          >
            10
          </Text>
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#212121",
                },
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            min.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginHorizontal": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#212121",
                },
                Object {
                  "color": "red",
                  "fontSize": 32,
                },
              ]
            }
          >
            14
          </Text>
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#212121",
                },
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            max.
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Temperaturen renders null for no temperaturen 1`] = `null`;
